File: ch1/check_permutation/checkPer.cc
#include <string>
#include <iostream>
using namespace std;

bool checkPermutations(string &str1, string str2){
    if(str1.length() != str2.length()){
        return false;
    }
    for(int i = 0; i < str1.length(); i++){
        for(int j = 0; j < str2.length(); j++){
            if(str1[i] == str2[j]){
                str2.erase(str2.begin() + j);
            }
        }
    }
    if(str2.length() > 0){
        return false;
    }
    return true;
}

int main(){
    string a = "abcda";
    string b = "bacda";
    string c = "abcde";
    cout << noboolalpha << checkPermutations(a, b) << endl;
    cout << noboolalpha << checkPermutations(a, c) << endl;
    return 0;
}

File: ch1/check_permutation/checkPer.py
import sys

def checkPermutations(str1, str2):
    if(len(str1) != len(str2)):
        return False
    else:
        for c in str1:
            if(str1.count(c) != str2.count(c)):
                return False
    return True

def main():
    if(len(sys.argv) != 3):
        print("Usage: $ python checkPer.py word1 word2")
    else:
        print(checkPermutations(sys.argv[1], sys.argv[2]))

if __name__ == "__main__":
    main()
